packages/twenty-server/src/modules/workflow/workflow-executor/workspace-services/workflow-executor.workspace-service.ts

```diff
@@ -23,43 +33,73 @@ export type WorkflowExecutorState = {
 };
 
 @Injectable()
-export class WorkflowExecutorWorkspaceService {
+export class WorkflowExecutorWorkspaceService implements WorkflowExecutor {
   constructor(
-    private readonly workflowActionFactory: WorkflowActionFactory,
     private readonly workspaceEventEmitter: WorkspaceEventEmitter,
     private readonly scopedWorkspaceContextFactory: ScopedWorkspaceContextFactory,
     private readonly workflowRunWorkspaceService: WorkflowRunWorkspaceService,
+    private readonly workflowActionExecutorWorkspaceService: WorkflowActionExecutorWorkspaceService,
   ) {}
 
   async execute({
     currentStepIndex,
     steps,
     context,
-    workflowExecutorState,
     attemptCount = 1,
     workflowRunId,
-  }: {
-    currentStepIndex: number;
-    steps: WorkflowAction[];
-    workflowExecutorState: WorkflowExecutorState;
-    context: Record<string, unknown>;
-    attemptCount?: number;
-    workflowRunId: string;
-  }): Promise<WorkflowExecutorState> {
+  }: WorkflowExecutorInput): Promise<WorkflowStepResult> {
     if (currentStepIndex >= steps.length) {
-      return { ...workflowExecutorState, status: WorkflowRunStatus.COMPLETED };
+      return {};
     }
 
     const step = steps[currentStepIndex];
 
-    const workflowAction = this.workflowActionFactory.get(step.type);
+    if (isWorkflowAction(step)) {
+      return this.executeAction({
+        currentStepIndex,
+        steps,
+        context,
+        attemptCount,
+        workflowRunId,
+      });
+    } else {
+      return this.executeStep({
+        currentStepIndex,
+        steps,
+        context,
+        attemptCount,
+        workflowRunId,
+      });
+    }
+  }
+
+  private async executeStep({
+    currentStepIndex,
+    steps,
+    context,
+    attemptCount = 1,
+    workflowRunId,
+  }: WorkflowExecutorInput) {
+    const step = steps[currentStepIndex];
 
-    const actionPayload = resolveInput(step.settings.input, context);
+    if (!isWorkflowStep(step)) {
+      throw new WorkflowExecutorException(
+        'Step is not a WorkflowStep',
+        WorkflowExecutorExceptionCode.INVALID_STEP,
+      );
+    }
 
-    let result: WorkflowActionResult;
+    const stepExecutor = this.getStepExecutor(step);
+    let result;
```

Comment:
The `result` variable was typed in the previous version. Why did you remove the explicit type?